### PR TITLE
🐞 fix: Datatable custom filter and other goodies

### PR DESCRIPTION
**Pull request by raxhvl**
## 🗒️ Description
Fixes:
-  Issues with custom fork and test filters.
-  Resolved a bug causing the datatable to break when changing entries per page.

Enhancements:

-  Upgraded jQuery from 3.5.1 to 3.7.1.
-  Upgraded DataTables from 1.10.21 to 2.1.7.
-  Migrated to minified scripts / css where possible for faster page load.
-  Added a plugin for searchable dropdowns.

image

## 🔗 Related Issues
#842 

## ✅ Checklist
- [x] All: Set appropriate labels for the changes.
- [x] All: Considered squashing commits to improve commit history.
- [x] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [x] ~~All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.~~ **skipped**
- [x] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [x] Tests: A PR with removal of converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been opened.
- [x] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [x] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

**Comment by danceratopz** [comment]
Hey @raxhvl, thanks for this! We had some issues with CI yesterday due to some upstream changes. Let me sort that out in [docs/include-all-parameters-in-test-parameter-datatables](https://github.com/ethereum/execution-spec-tests/tree/docs/include-all-parameters-in-test-parameter-datatables).

**Comment by raxhvl** [comment]
Just noticed that filters doesn't play nice with the search. Looking into it.

File: docs/javascripts/site.js [hunk 1]
```diff
@@ -1,92 +1,101 @@
...
-}
+// Config
+const FILTER_INPUT_SELECTOR = ".custom_dt_filter";
+let table;
 
-// Function to attach event listeners to the selectors
-function attachFilterListeners() {
+// Subscribe to when a page is loaded in mkdocs.
+// see: https://github.com/squidfunk/mkdocs-material/issues/5816#issuecomment-1667654560
+document$.subscribe(() => {
```
Comment (danceratopz) [inline docs/javascripts/site.js:7]: This breaks the data table initialization in the standalone pages as `site.js` also gets included there:
https://github.com/raxhvl/execution-spec-tests/blob/0c2549d6fc42a59b6a338e3f8821da722ce95827/docs/templates/function.html.j2#L24

**Review by danceratopz — COMMENTED** [review]
Thanks, this looks excellent so far! Much cleaner handling of the data table initialization. I really like the select styling. Just one small comment regarding the standalone html pages.

**Comment by raxhvl** [comment]
@danceratopz I think you will have to merge main again. Now search works within filter which is nice. Plus small updates to the UI. I'm done unless you have feedback.

File: docs/templates/function_parameter_datatable.html.j2 [hunk 2]
```diff
@@ -1,22 +1,47 @@
...
-    {% endfor %}
-</select>
-
-<table id="test_table" class="display nowrap">
+<div class="control_panel">
+    <div class="panel_row filters">
+    <div class="filter_wrapper">
+        <label for="fork_selector">
+         <span class="twemoji">
+            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
```
Comment (danceratopz) [inline docs/templates/function_parameter_datatable.html.j2:6]: Here's how this renders on the standalone page (Firefox):
image

**Review by danceratopz — CHANGES_REQUESTED** [review]
Thanks @raxhvl! The tables within the mkdocs pages look wonderful - love it!

Just one comment on the standalone pages. If there's no easy fix, we could consider splitting `function_parameter_datatable.html.j2` into a base template that would contain almost the logic and two derived pages, one for mkdocs, one for standalone. The only difference between the two would be whether to include the SVG, or not.

**Comment by raxhvl** [comment]
> Thanks @raxhvl! The tables within the mkdocs pages look wonderful - love it!
> 
> Just one comment on the standalone pages. If there's no easy fix, we could consider splitting `function_parameter_datatable.html.j2` into a base template that would contain almost all the logic and add two derived templates, one for mkdocs, one for standalone. The only difference between the two would be whether to include the SVG, or not.

It was a small fix - I copied over relevant css code from mkdocs to custom.css so it applies at both places now. Also, the standalone page was too wide and close to the browser boundary; I made it a little more compact - let me know if that looks okay

**Pull request merged**